app/models/story.rb

```diff
@@ -229,10 +230,16 @@ def self.find_similar_by_url(url)
     end
     urls = urls2.uniq
 
+    # trailing pound
+    urls.each do |u|
+      urls_with_trailing_pound.push u + "#"
+    end
+
     # if a previous submission was moderated, return it to block it from being
     # submitted again
     Story
       .where(:url => urls)
+      .or(Story.where("url RLIKE ?", urls_with_trailing_pound.join(".|")))
```

Comment:
What is the `.` for in this `join`?

Reply:
Short answer: 

Pattern matching: `.` matches any character.

Long answer:

So, let's say Alice submits the following URL https://artagnon.com/articles/leancoq#main, three minutes later Bob tries submitting https://artagnon.com/articles/leancoq

How do I know if https://artagnon.com/articles/leancoq has been submitted with an anchor before, i.e. https://artagnon.com/articles/leancoq#main? I first [added a `#` to all URLs](https://github.com/lobsters/lobsters/pull/802/files#diff-8a4ad5774f65a8a217b216fa2038a5bcR235), then I try to find any URL that matches the following pattern; https://artagnon.com/articles/leancoq#anycharacter

I'm able to do so by appending a `.` at the end of each URL.

As I pointed out [here](https://github.com/lobsters/lobsters/issues/801#issuecomment-571098708) I have my doubts about this approach.